SwissTransferFeatures/NewTransferView/NewTransferManager.swift

```diff
@@ -0,0 +1,260 @@
+/*
+ Infomaniak SwissTransfer - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import InfomaniakCore
+import OSLog
+import SwiftUI
+import SwissTransferCore
+
+private enum TmpDirType: String {
+    case all
+    case cache
+    case upload
+
+    var directory: URL {
+        get throws {
+            switch self {
+            case .all:
+                return FileManager.default.temporaryDirectory
+            case .cache:
+                return try URL.tmpCacheDirectory()
+            case .upload:
+                return try URL.tmpUploadDirectory()
+            }
+        }
+    }
+}
+
+@MainActor
+class NewTransferManager: ObservableObject {
+    @Published var uploadFiles = [UploadFile]()
```

Comment:
This could be private and not published ?
```suggestion
    private var uploadFiles = [UploadFile]()
```